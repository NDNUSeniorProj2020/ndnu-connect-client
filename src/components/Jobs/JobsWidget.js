import React from 'react';
import PropTypes from 'prop-types';
import { Card } from 'antd';

import './Jobs.css';
import JobCard from './JobCard';

export default function JobsWidget({ jobs }) {
  return (
    <div>
      <Card title="Jobs & Interships">
        {jobs ? jobs.map(job => (
          <div key={job.id}>
            <JobCard job={job} />
          </div>
        )) : <p>No jobs to display.</p>}
      </Card>
    </div>
  );
}

JobsWidget.propTypes = { jobs: PropTypes.array };
JobsWidget.defaultProps = { jobs: [] };
