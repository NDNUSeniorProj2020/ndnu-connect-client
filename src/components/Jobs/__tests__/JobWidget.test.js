import React from 'react';
import { shallow } from 'enzyme';

import JobsWidget from '../JobsWidget';

// Dummy data
const jobs = [
	{
		"id": 1,
		"title": "test1",
		"description": "description test",
		"qualifications": "bs degree",
		"pay": "1234",
		"link": "google.com",
		"date": "2020-03-26T11:59:26.227112-07:00",
		"type": "FULL",
		"user": 3
	}
];

describe('tests for JobWidget', () => {
	describe('snapshot tests', () => {
		it('renders without crashing', () => {
			const tree = shallow(<JobsWidget />);
			expect(tree).toMatchSnapshot();
		});

		it('renders with jobs', () => {
			const tree = shallow(<JobsWidget jobs={jobs} />);
			expect(tree).toMatchSnapshot();
		});
	});
});
