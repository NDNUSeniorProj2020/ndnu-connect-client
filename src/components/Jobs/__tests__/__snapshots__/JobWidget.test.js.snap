// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for JobWidget snapshot tests renders with jobs 1`] = `ShallowWrapper {}`;

exports[`tests for JobWidget snapshot tests renders without crashing 1`] = `ShallowWrapper {}`;
